genienlp/data_manipulation_scripts/clean_paraphrasing_dataset.py

```diff
@@ -32,31 +35,53 @@ def main():
                         help='The path to the input .tsv file.')
     parser.add_argument('output', type=str,
                         help='The path to the output .txt file.')
-    parser.add_argument('--start_special_token', type=str, default='<paraphrase>',
-                        help='The special token for the start of paraphrases.')
-    parser.add_argument('--end_special_token', type=str, default='</paraphrase>',
-                        help='The special token for the end of paraphrases.')
+
+    # By default, we swap the columns so that the target of paraphrasing will be a grammatically correct sentence, i.e. written by a human, not an NMT
     parser.add_argument('--first_column', type=int, default=1, help='The column index in the input file to put in the first column of the output file')
     parser.add_argument('--second_column', type=int, default=0, help='The column index in the input file to put in the second column of the output file')
+    
     parser.add_argument('--min_length', type=int, default=30, help='Minimum number of characters that each phrase should have in order to be included')
     parser.add_argument('--max_length', type=int, default=150, help='Maximum number of characters that each phrase should have in order to be included')
     parser.add_argument('--skip_check', action='store_true', help='Skip validity check.')
+    parser.add_argument('--skip_normalization', action='store_true', help='Do not remove quotation marks or detokenize.')
+    parser.add_argument('--lower_case', action='store_true', help='Convert everything to lower case.')
+    parser.add_argument('--max_output_size', type=int, default=1e10, help='Maximum number of examples in the output.')
 
     args = parser.parse_args()
 
     drop_count = 0
     # number_of_lines = get_number_of_lines(args.input)
+    # number_of_lines = get_number_of_lines(args.input)
+    output_size = 0
     with open(args.input, 'r') as input_file, open(args.output, 'w') as output_file:
+        writer = csv.writer(output_file, delimiter='\t')
         reader = csv.reader(input_file, delimiter='\t')
```

Comment:
Actually, you cannot use the CSV module to read our datasets, because they are not proper TSVs: fields are not delimited with `"`, and `"` is not escaped when it appears in a field. You should should split on `\t` directly.

Reply:
With `delimiter='\t'`, csv reader behaves more or less like spitting on `\t`. It has not had any issues with `"`, but it might for special cases. I'll fix this.